### PR TITLE
Llama fine-tuning accuracy fix

### PR DESCRIPTION
**Pull request by ugolowic**
This PR introduces a small change to loss calculation in GaudiTrainer, as well as some readability improvements.

After upgrade to transformers 4.49, there was accuracy drop (~7%) in fine-tuning workload involving llama3.1 70B (see example command below). It's related to how loss is calculated when `num_items_in_batch` is passed to `self.training_step()`.

The workload in question:
```
MASTER_ADDR=localhost MASTER_PORT=12345 PT_HPU_LAZY_MODE=1 \
    python3 ./examples/gaudi_spawn.py --world_size 8 --use_deepspeed \
    ./examples/language-modeling/run_lora_clm.py \
    --dataset_name tatsu-lab/alpaca \
    --bf16 True \
    --dataset_concatenation \
    --max_seq_len 4096 \
    --validation_split_percentage 4 \
    --save_strategy no \
    --warmup_ratio 0.03 \
    --lr_scheduler_type \"cosine\" \
    --logging_steps 10 \
    --do_train \
    --use_habana \
    --throughput_warmup_steps 3 \
    --use_flash_attention True \
    --flash_attention_causal_mask True \
    --num_train_epochs 1 \
    --eval_delay 2 \
    --do_eval \
    --lora_rank 4 \
    --fp8 False \
    --lora_target_modules \"q_proj\" \"v_proj\" \"k_proj\" \"o_proj\" \
    --per_device_eval_batch_size 1 \
    --attn_softmax_bf16 True \
    --output_dir /tmp/llama2_70b_hf_g2 \
    --eval_strategy epoch \
    --dataloader_num_workers 4 \
    --pipelining_fwd_bwd \
    --learning_rate 0.0018 \
    --deepspeed ./examples/language-modeling/llama3_ds_zero3_4k.json \
    --gradient_accumulation_steps 32 \
    --flash_attention_recompute True \
    --context_parallel_size 2 \
    --per_device_train_batch_size 1 \
    --model_name_or_path meta-llama/Llama-3.1-70B \
    --use_zero3_leaf_promotion \
    --cache_size_limit 256
```

### CODE REVIEW

**Comment by yafshar** [comment]
@ugolowic you are diverging from upstream transformers by removing `num_items_in_batch` not passed to `self.training_step()`. This would cause more issues in a long run rather than fixing the llama accuracy issue. If your argument is true, please make a PR in upstream transformers first.
